### PR TITLE
Implement mounting and unmounting snapshots

### PR DESCRIPTION
**Pull request by mkpankov**
Signed-off-by: Michael Pankov <work@michaelpankov.com>

---
This change is [Reviewable](https://reviewable.io/reviews/whamcloud/integrated-manager-for-lustre/2109)

### CODE REVIEW

File: iml-agent/src/action_plugins/lustre/snapshot.rs [hunk 1]
```diff
@@ -45,3 +45,29 @@ pub async fn destroy(d: Destroy) -> Result<(), ImlAgentError> {
...
+        &snapshot_name,
+    ];
+    lctl(args).await.map(drop)
+}
+
+pub async fn unmount(
+    (filesystem_name, snapshot_name): (String, String),
+) -> Result<(), ImlAgentError> {
+    let args = &[
+        "snapshot_umount",
```
Comment (ip1981) [inline iml-agent/src/action_plugins/lustre/snapshot.rs:66]: I wonder if we should stick to the lctl naming (umount/unmount) :)

Comment (mkpankov) [reply iml-agent/src/action_plugins/lustre/snapshot.rs:66]: I don't want these Unixy 1-letter contractions in our code to be honest

File: iml-agent/src/cli.rs [hunk 2]
```diff
@@ -699,6 +703,14 @@ async fn main() -> Result<(), Box<dyn std::error::Error>> {
             if let Err(e) = match command {
                 SnapshotCommand::Create(c) => lustre::snapshot::create(c).await,
                 SnapshotCommand::Destroy(d) => lustre::snapshot::destroy(d).await,
+                SnapshotCommand::Mount {
+                    fsname: fsn,
+                    name: n,
```
[the remark this answers is not in the document]
Comment (mkpankov) [reply iml-agent/src/cli.rs]: Thanks

**Comment by jgrund** [comment]
This patch has conflicts with master

File: iml-wire-types/src/snapshot.rs [hunk 3]
```diff
@@ -34,3 +34,23 @@ pub struct Destroy {
...
+    /// Filesystem name
+    pub fsname: String,
+    /// Snapshot name
+    pub name: String,
+}
+
+#[derive(serde::Deserialize, Debug)]
+#[cfg_attr(feature = "cli", derive(StructOpt))]
+/// Ask agent to unmount the snapshot
+pub struct Unmount {
```
Comment (ip1981) [inline iml-wire-types/src/snapshot.rs:51]: Btw, I was already thinking that simple tuple was good :) At least the same structure could be used for both mounting and unmounting.

Comment (jgrund) [reply iml-wire-types/src/snapshot.rs:51]: Tuple is good, but would need to be a tuple struct. The value here is being able to have the caller break at compile time if this changes.

**Pull request merged**